makeflow/src/makeflow.c

```diff
@@ -828,6 +849,31 @@ static void makeflow_node_complete(struct dag *d, struct dag_node *n, struct bat
 				makeflow_log_file_state_change(d, f, DAG_FILE_STATE_COMPLETE);
 		}
 
+		/* store node into archiving directory  */
```

Comment:
I'm concerned that this code is repeating the sequence of events necessary to fully expand the command string and file lists, which normally happens elsewhere.  That will be fragile if/when those sequences changes.  Can you find a way instead to record and save the command and files as they existed at the point of submission?